Makefile

```diff
@@ -111,7 +111,7 @@ vet: ## Run go vet against code.
 
 .PHONY: test
 test: manifests generate fmt vet envtest ## Run tests.
-	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) --bin-dir $(LOCALBIN) -p path)" go test $$(go list ./... | grep -v /e2e) -coverprofile cover.out
+	OPERATOR_NAMESPACE="$(NAMESPACE)" KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) --bin-dir $(LOCALBIN) -p path)" go test $$(go list ./... | grep -v /e2e) -coverprofile cover.out
```

Comment:
This will always overwrite the OPERATOR_NAMESPACE env var, which means that we are breaking existing functionality.

Consider the following command:
```
export OPERATOR_NAMESPACE=my-namespace
make test
```

or even
```
OPERATOR_NAMESPACE=my-namespace make test
```

In both cases, the operator will not get tested in "my-namespace" which is a breaking of the existing behavior

Reply:
@nb-ohad i dont think anything is working as existing functionality, what is the use of allowing a different name other than the `NAMESPACE` one?

Reply:
@nb-ohad  wrote:
> This will always overwrite the OPERATOR_NAMESPACE env var, which means that we are breaking existing functionality.
> 
> Consider the following command:
> 
> ```
> export OPERATOR_NAMESPACE=my-namespace
> make test
> ```
> 
> or even
> 
> ```
> OPERATOR_NAMESPACE=my-namespace make test
> ```
> 
> In both cases, the operator will not get tested in "my-namespace" which is a breaking of the existing behavior

Excellent point!

This is exactly why I wanted to set the env var only conditionally, when not already set. I used a new Makefile variable to circumvent syntax issues.

Upon @Madhu-1 's request, I changed it to set unconditionally.

I can try again to set it conditionally in the  test target definition without introducing the extra variable.

Reply:
@Madhu-1 I am using is locally when running multiple make test at the same time

Reply:
why not set `NAMSPACE` if you are using makefile to run the tests? i don't like to have specific variables for the test as we already have a variable for the operator namespace(kustomize namespace).

Reply:
@Madhu-1  wrote:
> why not set `NAMSPACE` if you are using makefile to run the tests? i don't like to have specific variables for the test as we already have a variable for the operator namespace(kustomize namespace).


do you mean using `make NAMESPACE=foo test` ?

I just confirmed that this works with the current patch.

Also `make NAMESPACE="" test`correctly triggers the panic. :) 

@nb-ohad  is this sufficient for your purpose?

Reply:
@Madhu-1 NAMESPACE affects other parts of the dependent make targets, I just want to let the operator know which namespace it is running on.

I am ok to defaulting to NAMESPACE if there is no OPERATOR_NAMESPACE defined. But I think the other way around is too limiting

Reply:
The latest update only sets the OPERATOR_NAMESPACE env var  (only in the test target) if it is not already set outside of make.
It avoids a new Makefile var.

Overriding it with

`OPERATOR_NAMESPACE=foo make test` and `make NAMESPACE=foo test` both weork. 

@Madhu-1 , @nb-ohad  - wdyt?

Reply:
LGTM